src/WebJobs.Script/Workers/Rpc/FunctionRpcService.cs

```diff
@@ -81,6 +81,12 @@ public override async Task EventStream(IAsyncStreamReader<StreamingMessage> requ
                     while (await messageAvailable());
                 }
             }
+            catch (Exception rpcException)
+            {
+                // We catch the exception, just to report it, then re-throw it
+                _logger.LogError(rpcException, "Exception encountered while listening to EventStream");
+                throw rpcException;
+            }
```

Comment:
just `throw;` to avoid resetting the stack. More info : https://stackoverflow.com/questions/730250/is-there-a-difference-between-throw-and-throw-ex !